### PR TITLE
Add goplugin KV generator example.

### PR DESCRIPTION
**Pull request by monopole**
This is covered by tests.

### CODE REVIEW

File: examples/kvSourceGoPlugin.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    - FRUIT=apple
+    - VEGETABLE=carrot
+  - pluginType: builtin
+    name: files
+    args:
+    - longsecret.txt
+  - pluginType: builtin
+    name: envfiles
+    args:
+    - foo.env
```
Comment (Liujingfang1) [inline examples/kvSourceGoPlugin.md]: This format is different from what's supported in 2.0.3. Considering users of kubectl(vendoring kustomize 2.0.3) will also look at this doc, can we use the 2.0.3 supported format?

File: examples/kvSourceGoPlugin.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+The problem that these three approaches share is that
+the purported secrets must live on disk, which adds
+additional security questions - who can see the files,
+who installs them, who deletes them, etc.
+
+
+## Secret values from anywhere
+
+A general alternative is to enshrine secret
```
Comment (Liujingfang1) [inline examples/kvSourceGoPlugin.md]: Can you add some explanation that generating Secrets via Go plugin works for Kustomize with version 2.0.4 and above.

File: examples/kvSourceGoPlugin.md (hunk 1 above)
Comment (monopole) [reply examples/kvSourceGoPlugin.md]: Good point - done.

File: examples/kvSourceGoPlugin.md (hunk 2 above)
Comment (monopole) [reply examples/kvSourceGoPlugin.md]: Done.

**Comment by Liujingfang1** [comment]
Not sure why the ci is hanging
/retest

**Comment by monopole** [comment]
everything passes locally.
travis says: waiting to queue.
shrug.  see where it is tomorrow.

**Comment by monopole** [comment]
I tested this to death locally, and will merge it since travis is having capacity issues.
from https://www.traviscistatus.com/
Update - We are bringing back capacity on both sites. 
Mar 28, 17:08 UTC
Update - We made great progress on cleaning up stale VMs in our infrastructure and we are preparing to add back capacity. 
Mar 28, 16:21 UTC
Update - We are stopping workers on travis-ci.com and travis-ci.org to hopefully alleviate the current situation. 
Mar 28, 15:18 UTC

**Pull request merged**